db/incidents.go

```diff
@@ -0,0 +1,35 @@
+package db
+
+import (
+	"github.com/flanksource/commons/logger"
+	"github.com/google/uuid"
+)
+
+func ReconcileIncidentStatus(incidentIDs []uuid.UUID) {
+	tx := Gorm.Exec(`
+        WITH evidences_agg as (
+            SELECT BOOL_AND(evidences.done) as done, incidents.status, incidents.id
+            FROM evidences
+            INNER JOIN hypotheses ON hypotheses.id = evidences.hypothesis_id
+            INNER JOIN incidents ON incidents.id = hypotheses.incident_id
+            WHERE
+                incidents.status != 'closed' AND
+                evidences.definition_of_done = true AND
+                incidents.id IN (?)
+            GROUP BY incidents.id
+        )
+        UPDATE incidents
```

Comment:
We probably need a trigger here to add the event = Or alternatively we just query the incidents that can change, and then update them individual + insert history item